mapie/tests/test_quantile_regression.py

```diff
@@ -519,6 +519,151 @@ def test_linear_regression_results(strategy: str) -> None:
+    ):
+        gb_trained1, gb_trained2 = clone(gb), clone(gb)
+        gb_trained1.fit(X_train, y_train)
+        gb_trained2.fit(X_train, y_train)
+        gb_trained3 = 3
+        list_estimators = [gb_trained1, gb_trained2, gb_trained3]
+        mapie_reg = MapieQuantileRegressor(
+            estimator=list_estimators,
+            cv="prefit",
+            alpha=0.3
+        )
+        mapie_reg.fit(
+            X_calib,
+            y_calib
+        )
+
+
+def test_non_trained_estimator() -> None:
+    """
+    Check that the estimators are all already trained when used in prefit.
+    """
+    with pytest.raises(
+        ValueError,
+        match=r".*instance is not fitted yet. Call 'fit' with appropriate*",
+    ):
+        gb_trained1, gb_trained2, gb_trained3 = clone(gb), clone(gb), clone(gb)
+        gb_trained1.fit(X_train, y_train)
+        gb_trained2.fit(X_train, y_train)
+        list_estimators = [gb_trained1, gb_trained2, gb_trained3]
+        mapie_reg = MapieQuantileRegressor(
+            estimator=list_estimators,
+            cv="prefit",
+            alpha=0.3
+        )
+        mapie_reg.fit(
+            X_calib,
+            y_calib
+        )
+
+
+def test_warning_alpha_prefit() -> None:
+    """
+    Check that the user is warned that the alphas need to be correctly set.
+    """
+    with pytest.warns(
+        UserWarning,
+        match=r".*WARNING: The alpha that is set needs to be the same*"
+    ):
+        gb_trained1, gb_trained2, gb_trained3 = clone(gb), clone(gb), clone(gb)
+        gb_trained1.fit(X_train, y_train)
+        gb_trained2.fit(X_train, y_train)
+        gb_trained3.fit(X_train, y_train)
+        list_estimators = [gb_trained1, gb_trained2, gb_trained3]
+        mapie_reg = MapieQuantileRegressor(
+            estimator=list_estimators,
+            cv="prefit",
+            alpha=0.3
+        )
+        mapie_reg.fit(
+            X_calib,
+            y_calib
+        )
+
+
+def test_prefit_and_non_prefit_equal() -> None:
+    """
+    Check that the user is warned that the alphas need to be correctly set.
+    """
+    list_estimators = []
+    alphas_ = [0.15, 0.85, 0.5]
+    for alpha_ in alphas_:
+        est = clone(qt)
+        params = {"quantile": alpha_}
+        est.set_params(**params)
+        est.fit(X_train, y_train)
+        list_estimators.append(est)
+    mapie_reg_prefit = MapieQuantileRegressor(
+        estimator=list_estimators,
+        cv="prefit",
+        alpha=0.3
+    )
+    mapie_reg_prefit.fit(X_calib, y_calib)
+    y_pred_prefit, y_pis_prefit = mapie_reg_prefit.predict(X)
+
+    mapie_reg = MapieQuantileRegressor(
+        estimator=qt,
+        alpha=0.3
+    )
+    mapie_reg.fit(X_train, y_train, X_calib=X_calib, y_calib=y_calib)
+    y_pred, y_pis = mapie_reg.predict(X)
+
+    np.testing.assert_allclose(y_pred_prefit, y_pred)
+    np.testing.assert_allclose(y_pis_prefit, y_pis)
```

Comment:
Add a last test check that MAPIE returns the same results if the 3 estimators are given as a list, tuple, or a np.array ?

Reply:
done